src/cmd/singularity/cli/pull.go

```diff
@@ -31,11 +43,23 @@ var PullCmd = &cobra.Command{
 	Args:   cobra.RangeArgs(1, 2),
 	PreRun: sylabsToken,
 	Run: func(cmd *cobra.Command, args []string) {
+		var uri, image string
+		image = ""
 		if len(args) == 2 {
-			libexec.PullImage(args[0], args[1], PullLibraryURI, force, authToken)
-			return
+			uri = args[1]
+			image = args[0]
+		} else {
+			uri = args[0]
+		}
+		BaseURI := strings.Split(uri, "://")
+		switch BaseURI[0] {
+		case SyCloudLibrary:
+			libexec.PullImage(image, uri, PullLibraryURI, force, authToken)
+		case Shub:
+			fmt.Println("Shub not yet supported")
```

Comment:
Why are we using `fmt.Println` here? We should be making use of `sylog` package across the board. `sylog.Errorf` or `sylog.Fatalf` should be used instead.